tests/jvm/src/test/scala-2.13/scala/meta/tests/testkit/prettyprinters/PrettyPrinterSuite.scala

```diff
@@ -44,4 +44,6 @@ class PrettyPrinterSuite extends FunSuite {
   checkOk("""1 join (())""")
 
   checkOk("""foo match{ case _ => _ => false}""")
+
+  checkOk("class Foo(`this`: Int)")
```

Comment:
what about `class Foo(foo: String) { def this(foo: Int) = this(foo.toString) }`?

Reply:
Well, it appears to work fine. Added a test case. It seems to parse it as `Term.This`, and it generates the output correctly.